### PR TITLE
Cowin integration updates

### PR DESCRIPTION
**Pull request by mkangia**
## Summary

:fish: 
Accommodate repeaters and generators to a real app. Originally added in https://github.com/dimagi/commcare-hq/pull/29949.

Agreed workflow:
Registration
1. HQ listens for "cowin_api_data" case, with "api" set to "registration", and sends a registration via API using info present on the case
2. Update to use new case property names, [reference](https://india.commcarehq.org/a/divax-covid-19-vax/reports/case_data/58b7ca88-7ea9-4f14-aedf-ae4daa22b1d8/)
3. Once registered, set `cowin_beneficiary_reference_id` on the person case. Person case id would be available on the "cowin_api_data" case itself

Vaccination update
1. HQ listens for "cowin_api_data" case, with "api" set to "vaccination", and send vaccination update to cowin API using info on the case. `cowin_beneficiary_reference_id` is available on the case itself (the app puts it on this case from the person case to avoid additional lookup in repeater).
2. Update to use new case property names, [reference](https://india.commcarehq.org/a/divax-covid-19-vax/reports/case_data/3dc0ff00-1694-4417-b4ff-7798cabbc20b/)
3. When updated, set "dose_{number}_notified" to True on the corresponding person case. Person case id should be available on the case itself.

These repeaters are not set for any project yet, so don't need to worry about supporting old case structures.

Side note: cowin_api_data cases are only used to store info related to api integration. They are then closed so that they are not loaded back on the phone to reduce load.
Hence, the repeater updates the "person" case so that app can use information on the person case, if needed, to show to the user.

## Feature Flag

`COWIN_INTEGRATION`

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

`custom/cowin/tests.py`

### Safety story

Added tests for changes.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by mkangia** [comment]
@kaapstorm i forgot to mark this ready for review. Just updated that status now.

**Review by kaapstorm — APPROVED** [review]

**Comment by kaapstorm** [comment]
Thanks @mkangia this is great.

**Comment by mkangia** [comment]
Thanks for the review @kaapstorm 

I could not test the changes with the sandbox apis because they have stopped working (that happens).
So i just made couple of changes to exactly match the api requirements, since I could not determine if it would be able to parse it on its own or not.

Going to merge post a test build but if you can take a look at these minor changes and share if there is any feedback please.

File: custom/cowin/tests.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def test_vaccination_payload(self, payload_doc_mock, connection_settings_mock):
                 "vaccine": "COVISHIELD",
                 "vaccine_batch": "123456",
                 "dose": 1,
-                "dose1_date": "2020-01-01",
+                "dose1_date": "01-01-2020",
```
Comment (kaapstorm) [inline custom/cowin/tests.py]: Guessing this is UK date format, not US date format? Why can't everyone just get onboard with ISO 8601?

Actually, just to make it more obvious, could you change the date to whatever 2020-12-31 is? (12-31-2020? 31-12-2020?)

**Review by kaapstorm — APPROVED** [review]

File: custom/cowin/tests.py (hunk 1 above)
Comment (mkangia) [reply custom/cowin/tests.py]: India format? DD-MM-YYYY :)

> Why can't everyone just get onboard with ISO 8601?

:(

Sure.

Comment (kaapstorm) [reply custom/cowin/tests.py]: Thanks for humouring me.

Yes! India format! South Africa format, Mexico format, France format ... nearly all of the countries ... nearly. :stuck_out_tongue_winking_eye:

**Pull request merged**